### PR TITLE
Add OpenSearch 1.3.6 and 2.4.0 to test matrix

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Review by wu-sheng — APPROVED** [review]
If e2e passed, I am good.

File: oap-server/server-library/library-elasticsearch-client/src/test/java/org/apache/skywalking/library/elasticsearch/ITElasticSearchTest.java [hunk 1]
```diff
@@ -104,6 +104,16 @@ public static Collection<Object[]> es() {
                                        "docker.elastic.co/elasticsearch/elasticsearch-oss"))
                     .withEnv("plugins.security.disabled", "true")
                     .withStartupTimeout(java.time.Duration.ofMinutes(5))
+            },
+            {
+                "OpenSearch 2.4.0",
```
Comment (wu-sheng) [inline oap-server/server-library/library-elasticsearch-client/src/test/java/org/apache/skywalking/library/elasticsearch/ITElasticSearchTest.java:109]: @wankai123 mentioned, we should consider to separating ElasticSearch IT from general ITs, and try to only booting ElasticSearch once per version.

Comment (wankai123) [reply oap-server/server-library/library-elasticsearch-client/src/test/java/org/apache/skywalking/library/elasticsearch/ITElasticSearchTest.java:109]: Yes, I found our ElasticSearch IT frequently fails recently and costs a lot of time.

Comment (wu-sheng) [reply oap-server/server-library/library-elasticsearch-client/src/test/java/org/apache/skywalking/library/elasticsearch/ITElasticSearchTest.java:109]: And IT had become the longest CI task in our e2e task list.

**Comment by wu-sheng** [comment]
> Error:  testDocUpdate[version: OpenSearch 2.4.0](org.apache.skywalking.library.elasticsearch.ITElasticSearchTest)  Time elapsed: 14.576 s  <<< ERROR!
java.util.concurrent.CompletionException: java.lang.RuntimeException: {"error":"no handler found for uri [/test-index-update/_doc/an id/_update] and method [POST]"}
Caused by: java.lang.RuntimeException: {"error":"no handler found for uri [/test-index-update/_doc/an id/_update] and method [POST]"}

We seem to have an incompatible issue.

File: oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/ElasticSearchVersion.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ private ElasticSearchVersion(final String distribution, final int major, final i
         this.minor = minor;
 
         if (distribution.equalsIgnoreCase("OpenSearch")) {
-            requestFactory = new V78RequestFactory(this);
+            requestFactory = new V81RequestFactory(this);
             codec = V78Codec.INSTANCE;
```
Comment (wu-sheng) [inline oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/ElasticSearchVersion.java:47]: V81 Factory with V78 code?

**Comment by wu-sheng** [comment]
BTW, let's put support versions of OpenSearch on the doc. https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-storage/#opensearch

File: oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/ElasticSearchVersion.java (hunk 2 above)
Comment (kezhenxu94) [reply oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/ElasticSearchVersion.java:47]: > V81 Factory with V78 code?

`RequestFactory` is to compose the API endpoint, such as `_doc/_update/<id>`, `<index>/_update/<id>`, etc., while `Codec` is to (de)serialize the request body, so they don't have to match in version. For example, they may change the endpoint from `_doc/_update/<id>` to `/_update/<id>` but the (de)serialization of the request body is unchanged.

**Review by wankai123 — APPROVED** [review]

**Pull request merged**